components/sup/src/manager/service/mod.rs

```diff
@@ -711,6 +716,7 @@ impl Service {
             }
         };
         self.last_health_check = Instant::now();
+        self.health_check = check_result;
```

Comment:
heh, yeah, this seems like something to keep track of 😄

Reply:
I ended up creating a `success` function for the `health_check` `enum` - happy to remove this and create the specific combination function if you think that is better.

I think it's pretty readable now though.